Source/WebCore/rendering/svg/SVGInlineTextBox.cpp

```diff
@@ -659,13 +660,23 @@ void SVGInlineTextBox::paintTextWithShadows(GraphicsContext& context, const Rend
                 if (paintingResourceMode().contains(RenderSVGResourceMode::ApplyToFill)) {
                     gradient = usedContext->fillGradient();
                     gradientSpaceTransform = usedContext->fillGradientSpaceTransform();
+                    pattern = usedContext->fillPattern();
                 } else {
                     gradient = usedContext->strokeGradient();
                     gradientSpaceTransform = usedContext->strokeGradientSpaceTransform();
+                    pattern = usedContext->strokePattern();
                 }
 
-                if (gradient) {
-                    usedContext->fillRect(textRootBlock->repaintRectInLocalCoordinates(), *gradient, gradientSpaceTransform);
+                ASSERT(!(gradient && pattern));
+                if (gradient || pattern) {
+                    if (gradient)
+                        usedContext->fillRect(textRootBlock->repaintRectInLocalCoordinates(), *gradient, gradientSpaceTransform);
+                    else {
+                        GraphicsContextStateSaver stateSaver(*usedContext);
```

Comment:
I would then check if a fillRect(FloatRect,Pattern&) exists to avoid that state swap

Reply:
Do you plan to take this in a separated PR?